lib/nerves_network.ex

```diff
@@ -26,11 +33,19 @@ defmodule Nerves.Network do
           | {:ipv4_subnet_mask, Types.ip_address()}
           | {:domain, String.t()}
           | {:nameservers, [Types.ip_address()]}
-          | {atom, any()}
+          | {:networks, [[wifi_settings]]}
+          | wifi_settings
 
   @typedoc "Keyword List settings to `setup/2`"
   @type setup_settings :: [setup_setting]
 
+  @typedoc "WiFi Settings"
+  @type wifi_settings ::
+          {:ssid, String.t()}
+          | {:key_mgmt, :"WPA-PSK" | :NONE}
+          | {:psk, String.t()}
+          | {:priority, non_neg_integer}
```

Comment:
Shouldn't this be `:non_neg_integer`? Like:

```suggestion
          | {:priority, :non_neg_integer}
```

Reply:
`non_neg_integer` is an elixir/dyalizer type. see: https://hexdocs.pm/elixir/typespecs.html#types-and-their-syntax

Reply:
Doh! I should have known that.